lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,278 @@
+// +build rpctest
+
+package itest
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+type chanFundMaxTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// amount is the amount in the Carol's wallet
+	amount btcutil.Amount
+
+	// pushAmt is the amount to be pushed to Dave
+	pushAmt btcutil.Amount
+
+	// feeRate is an optional fee in satoshi/bytes used when opening a channel
+	feeRate int64
+
+	// carolBalance is Carol's expected balance in her channel before
+	// subtracting the reserved amount to pay for a commitment transaction
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail
+	chanOpenShouldFail bool
+}
+
+func testChannelFundMax(net *lntest.NetworkHarness, ht *harnessTest) {
+
+	ctxb := context.Background()
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	carol, err := net.NewNode("carol", nil)
+	if err != nil {
+		ht.Fatalf("unable to start new node: %v", err)
+	}
+	defer shutdownAndAssert(net, ht, carol)
+
+	dave, err := net.NewNode("dave", nil)
+	if err != nil {
+		ht.Fatalf("unable to start new node: %v", err)
+	}
+	defer shutdownAndAssert(net, ht, dave)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	err = net.EnsureConnected(ctxt, carol, dave)
+	if err != nil {
+		ht.Fatalf("unable to connect peers: %v", err)
+	}
+
+	// Helper function that returns the fee estimate used for a tx
+	// with the given number of inputs and the optional change output.
+	// This matches the estimate done by the wallet.
+	fundingFee := func(numInput int, change bool) btcutil.Amount {
+		var (
+			feeRate        = chainfee.SatPerKWeight(12500)
```

Comment:
`feeRate` can be defined as a package level variable, it's never changed here. Then the var statement can become one line.